rasa/nlu/featurizers/count_vectors_featurizer.py

```diff
@@ -274,9 +419,26 @@ def persist(self, file_name: Text, model_dir: Text) -> Optional[Dict[Text, Any]]
         """
 
         file_name = file_name + ".pkl"
-        if self.vectorizer:
+        any_model_trained = False
+        for attribute in MESSAGE_ATTRIBUTES:
+            any_model_trained = self.is_trained[attribute] or any_model_trained
+
+        if any_model_trained:
             featurizer_file = os.path.join(model_dir, file_name)
-            utils.json_pickle(featurizer_file, self.vectorizer.vocabulary_)
+            if not self.use_shared_vocab:
+                utils.json_pickle(
+                    featurizer_file,
+                    [
```

Comment:
When vocab is shared, the persisted object is dict(`vocabilary_`), since it's a single vocabulary from single model. 
In `load()`, the cue for sharing of vocabulary is whether the retrieved object was a list or a dict.
If we make dict otherwise too, we would have to check if only one of the key has vocabulary set and others are None because there is not point storing the shared vocabulary for all attributes. So in any case there is some kind of a if construct.

Reply:
I see, but I still like the dict a bit better, rather than relying on the order in the list. But we always persist a config, so actually `meta["share_vocab"]` should give us needed info

Reply:
Changed to dict.